### PR TITLE
Bug 1152675 - Bluetooth v2, JS error while goes into settings::bluetooth panel, r=arthur

### PR DESCRIPTION
**Pull request by ian-liu**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bf7feefc4f661470eb8321f4f895ac01cf8916a3

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2f618f90a1dcfcfc2e25de5cde953b6c76c602c1

File: apps/settings/js/panels/bluetooth/panel.js [hunk 1]
```diff
@@ -193,6 +195,8 @@ define(function(require) {
           BtContext.stopDiscovery();
         }
 
+        BtContext.unobserve('discoverable', this._boundUpdateVisibilityTimer);
```
Comment (crh0716) [inline apps/settings/js/panels/bluetooth/panel.js]: I said un-observe the handler in `hide` just means that we should un-register it when leaving the panel. If you already un-observe other funcitons in `onBeforeHide`, let's do it there in one place.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fdb3bab3ceefba4c0f84b22d71d4cbaa6d8762e6

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=554861050448440efebdb39c02daa77a5cfd1512

**Pull request merged**